packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -320,6 +322,55 @@ describe('<DataGrid /> - Pagination', () => {
         );
         expect(getColumnValues(0)).to.deep.equal(['7', '8']);
       });
+
+      it('should update the amount of rows rendered and call onPageSizeChange when changing the table height', async () => {
+        const onPageSizeChange = spy();
+
+        const TestCaseAutoPageSize = (props: { nbRows: number; height?: number }) => {
+          const data = useData(props.nbRows, 10);
+
+          return (
+            <div style={{ width: 300, height: props.height }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
+                pagination
+                onPageSizeChange={onPageSizeChange}
+              />
+            </div>
+          );
+        };
+        const nbRows = 27;
+        const heightBefore = 780;
```

Comment:
Try to decompose this value so we know what to update if some height is changed in the future. Take as example:

```jsx
const heightBefore = 10 * rowHeight + headerHeight + footerHeight + border * 2; // 10 rows per page
```

Reply:
I'm doing the opposite
Setting a arbitrary height and calculating the amount of rows to expect
```ts
        const expectedViewportRowsLengthBefore = Math.floor(
          (heightBefore - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
            DEFAULT_GRID_OPTIONS.rowHeight,
        );
```
Which is the approach used in the tests above.